### PR TITLE
add hide angeltype in shift view option for angeltypes

### LINKED ISSUE #1164: Hide angeltype in shift view
for some angeltypes it would be usefull to hide them on shiftview if you are not part of the angeltype

### PR DESCRIPTION
**Pull request by xuwhite**
fixes #1164

### CODE REVIEW

File: includes/pages/user_shifts.php [hunk 1]
```diff
@@ -197,8 +198,11 @@ function load_types()
                 ON (
                     `user_angel_type`.`angel_type_id`=`angel_types`.`id`
                     AND `user_angel_type`.`user_id`=?
-                )
-            ORDER BY `angel_types`.`name`
+                )'
+            . ($isShico ? '' :
+            'WHERE angel_types.hide_on_shift_view = 0
+            OR user_angel_type.user_id IS NOT NULL ') .
```
Comment (MyIgel) [inline includes/pages/user_shifts.php:204]: this indentation does not feel right, imho some spaces should be added here?

File: includes/view/AngelTypes_view.php [hunk 2]
```diff
@@ -129,6 +131,15 @@ function AngelType_edit_view(AngelType $angeltype, bool $supporter_mode)
             $supporter_mode
                 ? form_info(__('Hide at Registration'), $angeltype->hide_register ? __('Yes') : __('No'))
                 : form_checkbox('hide_register', __('Hide at Registration'), $angeltype->hide_register),
+            $supporter_mode
+                ? form_info(__('angeltypes.hide.on.shift.view'), $angeltype->hide_on_shift_view ? __('Yes') : __('No'))
```
Comment (MyIgel) [inline includes/view/AngelTypes_view.php:135]:
```suggestion
                ? form_info(__('angeltypes.hide_shift_view'), $angeltype->hide_on_shift_view ? __('Yes') : __('No'))
```

Comment (MyIgel) [reply includes/view/AngelTypes_view.php:135]: or at least replace the dots with underscores ^^

File: resources/lang/de_DE/default.po [hunk 3]
```diff
@@ -2397,6 +2395,13 @@ msgstr "DECT"
 msgid "angeltypes.email"
 msgstr "E-Mail"
 
+msgid "angeltypes.hide.on.shift.view"
+msgstr "Auf Schicht Ansicht ausblenden"
```
Comment (MyIgel) [inline resources/lang/de_DE/default.po:2399]: Eventually
```suggestion
msgstr "Auf Schicht-Ansicht ausblenden"
```
(and other occurrences too)

File: db/factories/AngelTypeFactory.php [hunk 4]
```diff
@@ -27,6 +27,7 @@ public function definition(): array
             'no_self_signup'          => $this->faker->boolean(),
             'show_on_dashboard'       => $this->faker->boolean(),
             'hide_register'           => $this->faker->boolean(),
+            'hide_on_shift_view'      => $this->faker->boolean(),
```
Comment (MyIgel) [inline db/factories/AngelTypeFactory.php:30]: I'm thinking of how to make this shorter and / or more precise (less "technical"), eventually `hide_shift_overview`? Or `hide_shift_list`? The `on` seems redundant (or we should rename the `hide_register` later too to `hide_on_registration` (or `signup` when we are on it). ^^

**Review by MyIgel — CHANGES_REQUESTED** [review]

File: db/factories/AngelTypeFactory.php (hunk 4 above)
Comment (xuwhite) [reply db/factories/AngelTypeFactory.php:30]: hide_on_shift_view!!!!!

File: resources/lang/de_DE/default.po (hunk 3 above)
Comment (xuwhite) [reply resources/lang/de_DE/default.po:2399]: done

File: includes/view/AngelTypes_view.php (hunk 2 above)
Comment (xuwhite) [reply includes/view/AngelTypes_view.php:135]: done

File: includes/pages/user_shifts.php (hunk 1 above)
Comment (xuwhite) [reply includes/pages/user_shifts.php:204]: please be more precise
done

**Pull request merged**